src/passes/OptimizeInstructions.cpp

```diff
@@ -1224,14 +1230,78 @@ struct OptimizeInstructions
             !EffectAnalyzer(getPassOptions(), features, left->right)
                .hasSideEffects()) {
           switch (left->op) {
-            //   (x > y) | (x == y)    ==>    x >= y
             case EqInt32: {
-              if (right->op == GtSInt32) {
-                left->op = GeSInt32;
-                return left;
+              switch (right->op) {
+                //   (x >  y) | (x == y)   ==>    x >= y
```

Comment:
these comments are a little confusing to me. `left->op` is the `==`, so shouldn't it be on the left?

Reply:
yes, rhs and lhs expressions should be swapped